README_COVERAGE.md

```diff
@@ -0,0 +1,77 @@
+Randoop coverage using the Defects4J test suites
+----------------
+Defects4J is a collection of reproducible bugs collected with the goal of
+advancing software testing research.
+(See README.md in this directory for more information.)
+
+This README describes some changes to and additional tools added to
+Defects4j to calculate Randoop code coverage over the Defects4j test suites.
+
+Defects4J contains 395 bugs from the following open-source projects:
+
+| Identifier | Project name         | Number of bugs |
+|------------|----------------------|----------------|
+| Chart      | JFreeChart           |  26            |
+| Closure    | Closure compiler     | 133            |
+| Lang       | Apache commons-lang  |  65            |
+| Math       | Apache commons-math  | 106            |
+| Mockito    | Mockito              |  38            |
+| Time       | Joda-Time            |  27            |
+
+Requirements
+----------------
+ - Java 1.7
+ - Perl >= 5.0.10
+ - Git >= 1.9
+ - SVN >= 1.8
+
+All bugs have been reproduced and triggering tests verified, using the latest
+version of Java 1.7. All suites except Mockito seem to work with Java 1.8;
+however, the default coverage script runs all suites so you will need to
+have Java 1.7 as your default.
+
+Getting started
+----------------
+1. Clone Defects4J:
+    - `git clone https://github.com/rjust/defects4j`
+
+2. Initialize Defects4J (download the project repositories and external libraries,
+which are not included in the git repository for size purposes and to avoid redundancies):
+    - `cd defects4j`
+    - `./init.sh`
+
+3. Add Defects4J's executables to your PATH:
+    - `export PATH=$PATH:"path2defects4j"/framework/bin`
+
+4. Verify you have the perl package for DBI access to CSV files:
+    - `perldoc DBD/CSV.pm`
+    If this produces the man page for DBM::CSV you are ok, if not
+    you must install the package:
+    - `cpan DBD::CSV`
+
+Running the coverage tests
+----------------
+5. Please note you must use Java 7; Java 8 will cause failures.
+
+6. Tell the tools which version of Randoop you wish to test:
```

Comment:
What's the use case for supporting different versions? Is it to 1) support the developers of the test generation tools to quickly evaluate a new version before enabling it as the default in Defects4J or 2) allow a user of Defects4J to determine the most effective version of a test generation tool for a given experiment? I think that a user mostly wants to experiment with the most recent version supported by Defects4J.

I am inclined to only support one version, which is configured in the init script. Changing the version of a test generation tool only requires a simple change in that script.